src/WebJobs.Script/Description/DotNet/NoOpFunctionMetadataResolver.cs

```diff
@@ -0,0 +1,79 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Reflection;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs.Script.Abstractions.Description;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Scripting;
+
+namespace Microsoft.Azure.WebJobs.Script.Description
+{
+    public sealed class NoOpFunctionMetadataResolver : IFunctionMetadataResolver
+    {
+        private readonly Lazy<Task<PackageRestoreResult>> _emptyPackageResult = new Lazy<Task<PackageRestoreResult>>(GetEmptyPackageResult);
+        private static NoOpFunctionMetadataResolver _functionMetadataResolverSingleton;
+
+        private NoOpFunctionMetadataResolver()
+        {
+        }
+
+        public static NoOpFunctionMetadataResolver GetInstance()
```

Comment:
Can you follow the pattern we use other places for this, like NullLinuxContainerActivityPublisher?